cetmix_tower_server/models/cx_tower_scheduled_task.py

```diff
@@ -0,0 +1,291 @@
+        "server_id",
+        string="Servers",
+    )
+    action = fields.Selection(
+        [("command", "Command"), ("plan", "Flight Plan")], required=True
+    )
+    command_id = fields.Many2one("cx.tower.command", string="Command")
+    plan_id = fields.Many2one(string="Flight Plan", comodel_name="cx.tower.plan")
+    is_running = fields.Boolean(default=False, readonly=True)
+    interval_number = fields.Integer(default=1, help="Repeat every x.")
+    interval_type = fields.Selection(
+        [
+            ("minutes", "Minutes"),
+            ("hours", "Hours"),
+            ("days", "Days"),
+            ("weeks", "Weeks"),
+            ("months", "Months"),
+        ],
+        string="Interval Unit",
+        default="months",
+    )
+    next_call = fields.Datetime(
+        string="Next Execution Date",
+        required=True,
+        default=fields.Datetime.now,
+        help="Next planned execution date for this task.",
+    )
+    last_call = fields.Datetime(
+        string="Last Execution Date", help="Previous time the task ran successfully."
+    )
+    custom_variable_value_ids = fields.One2many(
+        "cx.tower.scheduled.task.cv",
+        "scheduled_task_id",
+        string="Custom Variable Values",
+    )
+    warning_message = fields.Text(
+        compute="_compute_warning_message",
+    )
+
+    # ---- Access. Add relation for mixin fields
+    user_ids = fields.Many2many(
+        relation="cx_tower_scheduled_task_user_rel",
+    )
+    manager_ids = fields.Many2many(
+        relation="cx_tower_scheduled_task_manager_rel",
+    )
+
+    @api.depends("interval_number", "interval_type")
+    def _compute_warning_message(self):
+        """
+        Show warning on the task form if interval in the scheduled task
+        is less than interval in the underlaying cron job.
+        """
+        cron = self.env.ref(
+            "cetmix_tower_server.ir_cron_run_scheduled_tasks", raise_if_not_found=False
+        )
+        if not cron:
+            self.warning_message = False
+            return
+
+        # Using now's date as the base point ensures a consistent and comparable
+        # reference when calculating the next scheduled execution for both the cron
+        # and the tasks.
+        now = fields.Datetime.now()
+        # _get_next_call is designed for tasks, but can also be used for the
+        # cron record, as both share the same interval fields. This keeps interval
+        # comparison logic consistent.
+        cron_next = self._get_next_call(cron, now)
+
+        for task in self:
+            task_next = self._get_next_call(task, now)
+            if task_next < cron_next:
+                task.warning_message = _(
+                    "Task interval is lower than underlying cron "
+                    "rate – may cause delays."
+                )
+            else:
+                task.warning_message = False
+
+    @api.constrains("interval_number")
+    def _check_interval_number(self):
```

Comment:
Wouldn't and SQL constraint be faster here? eg

```
interval_positive CHECK (rec.interval_number > 0)
```

Reply:
done